### PR TITLE
Add date range filters

### PR DESCRIPTION
**Pull request by Rayhatron**
This PR adds the ability to filter the posts returned by the PostSelect component based on selected dates. More details about context and technical brief can be found in the issue #89.

### Testing instructions

- Checkout the branch 89-add-date-range-filters
- Add an extra prop ```showDateFilters``` to the PostSelect component in your block
- Edit a post and add your block to the content so that the PostSelect component is shown
- Click the Select button to open the post select modal
- Two new buttons should show up in two columns titled "Published after date" and "Published date before" respectively
- Click the button below either of the columns to se the appropriate date
- Click "Filter posts" button at the bottom of modal screen and the posts returned should filter to include only those in the selected date range
- NB: if you set "Published after date" to say Thursday 4 November then posts published on the day will be included. Open to discussing if they shouldn't be included such that only posts truly published after that day would be returned.
- Also test that the "Published after date" cannot be after the "Published before date" and the reverse of the "Published before date" cannot be before the "Published after date" when that validation fails, an error notice should be shown.

### CODE REVIEW

**Comment by kadamwhite** [comment]
image

**Review by kadamwhite — CHANGES_REQUESTED** [review]
This is great work! I think there's a couple areas where the interaction could be streamlined. Most of all, the fact that you select the two dates separately and need to know to "x" out of the modal after selecting, feels a bit awkward to me.

What do you think about having both the start and end dates in one modal, so that the UI above the filters would say
image
(or "between X & Y" or "before X", though that admittedly adds a lot more translatable strings)

and then batch the start and end dates UI within a modal that has a clear confirmation button:
image

Validation errors could also be displayed within this modal, rather than hidden in the parent modal.

If this is out of scope given available time, we can roll the UI you have already and iterate later. However I'm curious what you (and anybody else reading) thinks.

File: js/post-select/containers/browse-filters.js
Comment (ivankristianto) [inline js/post-select/containers/browse-filters.js:31]: Passing down this props all the way from the root component till the date component seems messy.
I wonder if we can use Context API here?
cc: @kadamwhite looking for your recommendation here too.

**Review by ivankristianto — COMMENTED** [review]
@Rayhatron So far so good, i left couple comments to improve as additional to Kadam's feedbacks.

File: js/post-select/containers/browse-filters.js
Comment (Rayhatron) [reply js/post-select/containers/browse-filters.js:31]: @kadamwhite I've implemented the UI/UX changes you shared and also fixed the bug when switching dates. The outstanding thing from the feedback is this one which was shared by @ivankristianto. I'm not quite sure how to go about it yet.

Comment (kadamwhite) [reply js/post-select/containers/browse-filters.js:31]: I like @ivankristianto's suggestion a lot, but since we're not using that currently, I feel that introducing Context here would feel out-of-scope for this ticket. We don't really want to have two competing paradigms at once: I'd rather proceed with the property passing method @Rayhatron is using, and then open a new ticket to refactor the components to use Context more heavily so that we can adjust things wholesale.

Comment (Rayhatron) [reply js/post-select/containers/browse-filters.js:31]: Makes sense, I've created the new ticket for this.

**Review by ivankristianto — APPROVED** [review]
Approve from me. But also suggest to wait for another approval from @kadamwhite

**Review by kadamwhite — APPROVED** [review]
Great cleanup, and works just as advertised!

Re: one of the notes in the description, I think the "inclusive" date behavior is good. The only problem i was able to find with the implementation as-is is that if you select the same date for the start and end of the date range, it doesn't correctly find posts on that day; but we can clean that up in a follow-up ticket. The overall filter behavior looks great.

**Pull request merged**
